### PR TITLE
Initial graphite tags support

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This PR adds initial Graphite tags support (issue #9230) by using new functions:
- seriesByTag()
- groupByTags()
- aliasByTags()

Also, this fixes graphite function editor issue #9238

### CODE REVIEW

File: public/app/plugins/datasource/graphite/gfunc.js [hunk 1]
```diff
@@ -822,6 +822,49 @@ function (_, $) {
...
+  addFuncDef({
+    name: 'seriesByTag',
+    category: categories.Special,
+    params: [
+      { name: "tagExpression", type: "string" },
+      { name: "tagExpression", type: "string", optional: true },
+      { name: "tagExpression", type: "string", optional: true },
+      { name: "tagExpression", type: "string", optional: true },
+    ],
+    version: '1.0'
```
Comment (torkelo) [inline public/app/plugins/datasource/graphite/gfunc.js]: version should be 1.1?

Comment (alexanderzobnin) [reply public/app/plugins/datasource/graphite/gfunc.js]: Yes, sure.

**Comment by alexanderzobnin** [comment]
@torkelo found another issue. Function `seriesByTag()` used as first function and in this case function parser shifts params:
```js
case 'bool':
      case 'string':
      case 'number':
        if ((index-1) >= func.def.params.length) {
          throw { message: 'invalid number of parameters to method ' + func.def.name };
        }
        this.addFunctionParameter(func, astNode.value, index, true); // always true
      break;
```
Can you explain why should we shift params?

**Comment by alexanderzobnin** [comment]
Seems, it's because function usually takes metric as the first argument.

**Comment by alexanderzobnin** [comment]
Added workaround for `seriesByTag()` function, please, check it.

**Comment by alexanderzobnin** [comment]
Another question: graphite `seriesByTag()` has special tag `name` which allows to select metric name. It's a bit obvious, maybe it makes sense to add this tag as first default param? Like:
```
seriesByTag(name=~.*)
```

**Comment by torkelo** [comment]
That could make sense. But could also be a bit dangerous as it would match to many series

**Comment by alexanderzobnin** [comment]
By default seriesByTag() returns all series with specified tags. But not sure, maybe `name=~.*` produce additional load to database. So better to use a more concrete name.

**Comment by torkelo** [comment]
even though we will handle seriesByTag differently I am merging this to get the fix  for https://github.com/grafana/grafana/issues/9238

**Comment by alexanderzobnin** [comment]
Ok

**Pull request merged**